app/components/Onboarding/Steps/ConnectionConfirm.js

```diff
@@ -58,10 +60,16 @@ class ConnectionConfirm extends React.Component {
       cert: connectionCert,
       macaroon: connectionMacaroon
     }
+    // if we have a connection string, parse the connection details from that.
     if (connectionString) {
       const { host, port, macaroon } = parseConnectionString(connectionString)
       options = { type: connectionType, host: `${host}:${port}`, macaroon }
     }
+    // Otherwise, if we have a connection uri, parse the connection details from that.
+    else if (connectionUri) {
+      const { host, cert, macaroon } = decode(connectionUri)
```

Comment:
We may want to use Error Boundaries here 
[https://reactjs.org/docs/error-boundaries.html](https://reactjs.org/docs/error-boundaries.html)
In case `decode` or `parseConnectionString` throw, which they, judging from their name could, the whole component tree will unmount and it will get ugly

Reply:
What I have pushed so far was just the result of hacking until I had something that works. This whole PR is being reworked. I've read about Error Boundaries  before but not implemented them. I'll see how I can work into this in order to prevent the type of possible issue you mentioned.

Reply:
Gotcha